fineract-provider/src/main/java/org/apache/fineract/cob/service/InlineLoanCOBExecutorServiceImpl.java

```diff
@@ -0,0 +1,137 @@
+    public CommandProcessingResult executeInlineCOB(JsonCommand command, String jobName) throws LoanAccountLockCannotBeOverruledException {
+        List<Long> loanIds = dataParser.parseExecution(command);
+        List<LoanAccountLock> loanAccountLocks = getLoanAccountLocks(loanIds);
+        execute(loanAccountLocks, loanIds, jobName);
+        return new CommandProcessingResultBuilder().withCommandId(command.commandId()).build();
+    }
+
+    public List<Long> getLoanIds() {
+        return loanIds;
+    }
+
+    private List<LoanAccountLock> getLoanAccountLocks(List<Long> loanIds) {
+        List<LoanAccountLock> loanAccountLocks = new ArrayList<>();
+        List<Long> alreadyLockedLoanIds = new ArrayList<>();
+        loanIds.forEach(loanId -> {
+            Optional<LoanAccountLock> loanLockOptional = loanAccountLockRepository.findById(loanId);
+            if (loanLockOptional.isPresent()) {
+                LoanAccountLock loanAccountLock = loanLockOptional.get();
+                if (isLockOverruleable(loanAccountLock)) {
+                    loanAccountLocks.add(loanAccountLock);
+                } else {
+                    alreadyLockedLoanIds.add(loanId);
+                }
+            } else {
+                loanAccountLocks.add(new LoanAccountLock(loanId, LockOwner.LOAN_INLINE_COB_PROCESSING));
+            }
+        });
+        if (!alreadyLockedLoanIds.isEmpty()) {
+            String message = "There is a hard lock on the loan account without any error, so it can't be overruled.";
+            String loanIdsMessage = " Locked loan IDs: " + alreadyLockedLoanIds;
+            throw new LoanAccountLockCannotBeOverruledException(message + loanIdsMessage);
+        }
+        return loanAccountLocks;
+    }
+
+    private void execute(List<LoanAccountLock> loanAccountLocks, List<Long> loanIds, String jobName) {
+        lockLoanAccounts(loanAccountLocks);
+        this.loanIds = loanIds;
+        Job inlineLoanCOBJob;
+        try {
+            inlineLoanCOBJob = jobLocator.getJob(jobName);
+        } catch (NoSuchJobException e) {
+            throw new JobNotFoundException(jobName, e);
+        }
+        JobParameters jobParameters = new JobParametersBuilder(jobExplorer).getNextJobParameters(inlineLoanCOBJob)
+                .addJobParameters(new JobParameters()).toJobParameters();
+        JobExecution jobExecution;
+        try {
+            jobExecution = jobLauncher.run(inlineLoanCOBJob, jobParameters);
+        } catch (Exception e) {
+            final String msg = "Job execution failed for job with name:" + jobName;
+            log.error("{}", msg, e);
+            throw new PlatformInternalServerException("error.msg.sheduler.job.execution.failed", msg, jobName, e);
+        }
+        if (!BatchStatus.COMPLETED.equals(jobExecution.getStatus())) {
+            final String msg = "Job execution failed for job with name:" + jobName;
+            log.error("{}", msg);
```

Comment:
What's the point of this? With the String concatenation above you're defeating the purpose of placeholders in the logger. Should inline the log message and pass the jobName as a parameter.